content/doc/book/glossary/index.adoc

```diff
@@ -5,15 +5,119 @@ layout: chapter
+    run. Artifacts are typically *archived* onto the Jenkins <<master,Master>> for
+    later retrieval by users.
+Build:: [[build]]
+    Result of a single execution of a <<project,Project>>
+Cloud:: [[cloud]]
+    A System Configuration which provides dynamic <<agent,Agent>>
+    provisioning and allocation, such as that provided by the
+    link:https://plugins.jenkins.io/azure-vm-agents[Azure VM Agents]
+    or
+    link:https://plugins.jenkins.io/ec2[Amazon EC2] plugins.
+Core:: [[core]]
+    The primary Jenkins application (`jenkins.war`) which provides
+    the basic web UI, configuration, and foundation upon which <<plugin, Plugins>>
+    can be built.
+Downstream:: [[downstream]]
+    A configured <<pipeline,Pipeline>> or <<project,Project>> which is triggered
+    as part of the execution of a separate Pipeline or Project.
+Executor:: [[executor]]
+    A thread for execution of <<pipeline,Pipelines>> or <<project,Projects>> on a
+    <<node, Node>>. A Node may have zero or more Executors configured which
+    corresponds to how many concurrent Projects or Pipelines are able to execute on
+    that Node.
+Fingerprint:: [[fingerprint]]
+    A hash considered globally unique to track the usage of an
+    <<artifact,Artifact>> or other entity across multiple
+    <<pipeline,Pipelines>> or <<project,Projects>>.
+Folder:: [[folder]]
+    An organizational container for <<pipeline,Pipelines>> and/or
+    <<project,Projects>> in the web UI, similar to folders on a file system.
+Item:: [[item]]
+    An entity in the web UI corresponding to either a:
+    <<folder,Folder>>, <<pipeline,Pipeline>>, or <<project,Project>>.
+Job:: [[job]]
+    A deprecated term, synonymous with <<project,Project>>.
+Master:: [[master]]
+    The central, coordinating process which stores configuration, loads plugins,
+    and renders the various user interfaces for Jenkins.
+Node:: [[node]]
+    A machine which is part of the Jenkins environment and capable
+    of executing <<pipeline,Pipelines>> or <<project,Projects>>. Both the
+    <<master,Master>> and <<agent,Agents>> are considered to be Nodes.
+Project:: [[project]]
+    A user configured description of work which Jenkins should perform, such as
+    building a piece of software, etc.
+Pipeline:: [[pipeline]]
+    A user-defined model of a continuous delivery pipeline, for more read the
+    <<pipeline#,Pipeline chapter>> in this handbook.
+Plugin:: [[plugin]]
+    An extension to Jenkins functionality provided separately
+    from Jenkins <<core,Core>>.
+Publisher:: [[publisher]]
+    Part of a <<build,Build>> after the completion of all configured
+    <<step,Steps>> which publishes reports, sends notifications, etc.
+Step:: [[step]]
+    A single task; fundamentally steps tell Jenkins _what_ to do inside of a
+    <<pipeline,Pipeline>> or <<project,Project>>.
+Trigger:: [[trigger]]
+    A criteria for queueing a new <<pipeline,Pipeline>> run or <<build,Build>>
+    of a <<project, Project>>.
```

Comment:
Probably want to mention the verb "Trigger" since it is used in talking about upstream and downstream.